### DISCUSSION TITLE
How to get a point time series data

### DISCUSSION THREAD

**Discussion by sunzhe-96** [gmtsar/gmtsar#272] [Q&A] [answered]
Hello,everyone,I want to get the continuous deformation value of a point in the final velocity result, how can I do that？

**Comment by Xiaohua-Eric-Xu** [comment, marked as answer]
There is a script that reads in GMTSAR called `extract_one_time_series.csh `, that'll read products from sbas in radar coordinates and form a time-series.

**Reply by sunzhe-96** [reply]
Hello,Eric,I am sorry to bother you,when I run the `extract_one_time_series.csh`,there seems to be some problem.
extract_one_time_series.csh 104.85 28.45 S1_20200515_ALL_F3.PRM dem.grd scene.tab 5 5
hermite: interpolation not in center interval
nan!
hermite: interpolation not in center interval
nan!
extracting at position range  and azimuth  ...
grid xinc: 8, yinc: 2; average half samp: 2 2 ...
sampling over 2/12/-2147483647/-2147483647 ...
Working on date disp_2019200.grd
grdcut [ERROR]: Syntax error -R option. Correct syntax:
	-R<xmin>/<xmax>/<ymin>/<ymax>[/<zmin>/<zmax>]
	  Append r if giving lower left and upper right coordinates
	-Rg or -Rd for global domain
	-R<grdfile> to take the domain from a grid file
grdcut [ERROR]: Offending option -R2/12/-2147483647/-2147483647
grdinfo [ERROR]: Error for input file: No such file (tmp_cut.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files
Working on date disp_2019236.grd
grdcut [ERROR]: Syntax error -R option. Correct syntax:
	-R<xmin>/<xmax>/<ymin>/<ymax>[/<zmin>/<zmax>]
	  Append r if giving lower left and upper right coordinates
	-Rg or -Rd for global domain
	-R<grdfile> to take the domain from a grid file
grdcut [ERROR]: Offending option -R2/12/-2147483647/-2147483647
grdinfo [ERROR]: Error for input file: No such file (tmp_cut.grd)
grdinfo [ERROR]: Syntax error: Must specify one or more input files

**Reply by Xiaohua-Eric-Xu** [reply]
Feels something's incorrect with your coverage. If you run`gmt grdinfo dem.grd`, what's the boundaries on your dem.grd?
Or if you run `echo 104.85 28.45 | gmt grdtrack -Gdem.grd`, what's the output?

**Reply by sunzhe-96** [reply]
> 
Thank you for your reminder!,Eric, I got the dem wrong and it works fine now.

**Reply by Gigi-gugu** [reply]
> Feels something's incorrect with your coverage. If you run`gmt grdinfo dem.grd`, what's the boundaries on your dem.grd? Or if you run `echo 104.85 28.45 | gmt grdtrack -Gdem.grd`, what's the output?
![log1](https://user-images.githubusercontent.com/92804122/188264646-bbdee418-960f-43c1-aa59-1ddd4d2be6e6.jpg)
![log2](https://user-images.githubusercontent.com/92804122/188264650-4216b61f-c53b-4f27-bf6f-df2c7e82d52e.jpg)
Hello, Eric

I think I have a similar problem. I think the error comes from (extracting at position range and azimuth...). According to your proposal run echo 117.114 39.257 | gmt grdtrack -Gdem.grd, but the output is empty. I'm sure my longitude and latitude are OK. Can you tell me where the problem comes from?

**Reply by Xiaohua-Eric-Xu** [reply]
Your dem.grd is likely the dem error output from sbas. You'll need the actual digital elevation model in your topo directory.

**Reply by Gigi-gugu** [reply]
> You
![log](https://user-images.githubusercontent.com/92804122/188779092-8af360f2-a466-4d09-a2c3-fe10ea939ba0.jpg)
r dem.grd is likely the dem error output from sbas. You'll need the actual digital elevation model in your topo directory.

This is another example of mine, but it works. So before I tested, I personally thought that the error should not be from SBAS  dem,grd. Because in this example, its dem.grd is also an error output.However, the dem.grd I copied from topo according to your opinion is OK. So I would like to ask whether extract_one_time_series. csh uses SBAS dem.grd or Topo dem.grd.

**Reply by Xiaohua-Eric-Xu** [reply]
It has to be topo dem.grd. We should change the output names in sbas to avoid confusion.

**Reply by Xiaohua-Eric-Xu** [reply]
Just did. https://github.com/gmtsar/gmtsar/commit/e5e1bb8e3108b3eee4eca13975f713c96108d79e

**Reply by Gigi-gugu** [reply]
Thank you very much for your help. I still have a question. Can you give me another answer? In the process of processing sequence, why are there some master.PRM and some supermaster.PRM? When only one IW is processed, there is not supermaster.PRM.

**Reply by Xiaohua-Eric-Xu** [reply]
the supermaster.PRM is a file produced for merging a stack of data. For a pair, even with all three subswaths processed, you only need to merge once using the master coordinates. However, for a stack of data, an interferogram is produced using one reference and one repeat image, both of which are aligned to a single master image. Thus we need a supermaster PRM file for merging.

**Comment by Gigi-gugu** [comment]
Good!Thank you very much! I wish you a Happy Mid Autumn Festival and a happy family reunion.

**Reply by Xiaohua-Eric-Xu** [reply]
Thanks!
